src/components/Data/Documents/ViewMap.vue

```diff
@@ -0,0 +1,209 @@
+          >
+          </l-marker>
+        </l-map>
+      </div>
+
+      <div v-if="currentDocument" class="col s2 document-info">
+        <div class="row">
+          <div class="col s9">
+            {{ currentDocument.id }}
+          </div>
+          <a
+            v-if="canEdit"
+            href=""
+            :title="canEdit ? 'Edit Document' : 'You are not allowed to edit this Document'"
+            @click.prevent="editCurrentDocument">
+            <i class="fa fa-pencil-alt" :class="{'disabled': !canEdit}"></i>
+          </a>
+
+          <dropdown :id="currentDocument.id" myclass="icon-black">
+            <li>
+              <a
+                v-bind:class="{'disabled': !canDelete}"
+                @click="deleteCurrentDocument()"
+                v-title="{active: !canDelete, title: 'You are not allowed to delete this document'}">
+                Delete
+              </a>
+            </li>
+          </dropdown>
+        </div>
+        <hr>
+        <p
+          v-json-formatter="{
+            content: {
+              id: currentDocument.id,
+              body: currentDocument.content
+            },
+            open: true
+          }"
+          />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { LMap, LTileLayer, LMarker, LPopup, LIconDefault } from 'vue2-leaflet'
+import L from 'leaflet'
+import Dropdown from '../../Materialize/Dropdown'
+import '../../../../src/assets/leaflet.css'
+import JsonFormatter from '../../../directives/json-formatter.directive'
+import title from '../../../directives/title.directive'
+import {
+  canEditDocument,
+  canDeleteDocument
+} from '../../../services/userAuthorization'
+
+export default {
+  name: 'ViewMap',
+  components: {
+    LMap,
+    LTileLayer,
+    LMarker,
+    LPopup,
+    LIconDefault,
+    Dropdown
+  },
+  props: {
+    documents: {
+      type: Array,
+      required: true
+    },
+    getCoordinates: {
+      type: Function,
+      required: true
+    },
+    index: String,
+    collection: String
+  },
+  directives: {
+    JsonFormatter,
+    title
+  },
+  data() {
+    return {
+      latField: null,
+      lngField: null,
+      map: null,
+      url: 'http://{s}.tile.osm.org/{z}/{x}/{y}.png',
+      attribution:
+        '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+      currentDocument: null,
+      defaultIcon: new L.Icon({
+        iconUrl: 'static/images/marker-icon-2x-blue.png',
+        shadowUrl: 'static/images/marker-shadow.png',
+        iconSize: [25, 41],
+        iconAnchor: [12, 41],
+        popupAnchor: [1, -34],
+        shadowSize: [41, 41]
+      }),
+      selectedIcon: new L.Icon({
+        iconUrl: 'static/images/marker-icon-2x-green.png',
+        shadowUrl: 'static/images/marker-shadow.png',
+        iconSize: [25, 41],
+        iconAnchor: [12, 41],
+        popupAnchor: [1, -34],
+        shadowSize: [41, 41]
+      })
+    }
+  },
+  computed: {
+    latLngs() {
```

Comment:
I'd keep the `coordinates` name on all the methods (`latLngs` and `getCoordinates` use two names for the same concept)